### PR TITLE
Update client aa1 to implement 2

### PR DESCRIPTION
**Pull request by schettn**
**Checklist:**

 
 - [ ] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [x] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Documentation (non-breaking change which adds documentation)
 - [x] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently the client scraper lacks of most http methods
- Currently there is no instagram client
- Currently there minor errors in session awareness.
- Currently it is not possible to upload files.
- Currently there are tasks for snek calls.

**What is the new behavior (if this is a feature change)?**
- Now it there is a new method which allows performing all http methods.
- Now there is a instagram client
- Now the session refresh caused by an invalid token is fixed.
- Now it is possible to upload files.
- Now there are only tasks which are required for the session awareness. All other interaction
   should be handled by the session runner. `src/session/sessions.ts` 

**Other information**:
- 🐍

### CODE REVIEW

File: src/session/sessions.ts [hunk 1]
```diff
@@ -141,6 +139,65 @@ class CookieSession extends Session {
...
+   * @author Nico Schett <contact@schett.net>
+   * @param {string} sId A session name
+   * @param {string} token A instagram access token
+   * @param {Endpoint} ep A endpoint
+   */
+  constructor(sId: string, token: string, public ep: ScraperEndpoint) {
+    super(sId);
+
+    this.tokenName = sId + "-" + this.tokenName;
+
```
Comment (pinterid) [inline src/session/sessions.ts]:
```suggestion
```

File: src/templates/instagram/tasks.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+//> Sessions
+// Contains the SNEK session
+import { InstagramSession } from "../../session/sessions";
+//> Instagram Endpoint paths
+import * as paths from "./paths";
```
Comment (pinterid) [inline src/templates/instagram/tasks.ts:6]:
```suggestion
//#region > Imports
//> Sessions
// Contains the SNEK session
import { InstagramSession } from "../../session/sessions";
//> Instagram Endpoint paths
import * as paths from "./paths";
//#endregion
```

File: src/templates/snek/tasks/auth.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    );
+
+    return response;
+  }
+
+  /**
+   * User login.
+   *
+   * @param {string} user A User defined by username and password
+   * @returns {Promise<ApolloResult<GraphQLAuthentication>>} Authentication data.
```
Comment (pinterid) [inline src/templates/snek/tasks/auth.ts]:
```suggestion
   * @returns {Promise<ApolloResult<GraphQLAuthentication>>} Authentication data
```

File: src/templates/snek/tasks/index.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+//> Queries
+import * as queriesUser from "../queries/user";
+//> Types
+// Contains the graphql response types
+import { ApolloResult, TaskTypes } from "../types";
+// Contains error handling for tasks
+import { TaskError } from "../errors";
+import AuthTask from "./auth";
+import UserTask from "./user";
+
```
Comment (pinterid) [inline src/templates/snek/tasks/index.ts:18]:
```suggestion
//#endregion

//#region > Classes
```

File: src/templates/snek/tasks/index.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      );
+      response = await this.session.ep.sendQuery<T>(query, variables);
+    }
+
+    this.handleErrors(response);
+
+    return response;
+  }
+}
+
```
Comment (pinterid) [inline src/templates/snek/tasks/index.ts:57]:
```suggestion
//#endregion

```

File: src/templates/snek/types.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
+/**
+ * All types
+ */
+
```
Comment (pinterid) [inline src/templates/snek/types.ts]: Missing region for these exports?

**Review by pinterid — CHANGES_REQUESTED** [review]
Some changes have to be done.

**Review by pinterid — APPROVED** [review]
LGTM

File: src/endpoints/scraper.ts [hunk 7]
```diff
@@ -123,7 +110,80 @@ class Scraper implements ScraperEndpoint {
...
+      if (!response.ok) {
+        //#ERROR
+        throw new Error(response.statusText);
+      }
+
+      return response.json().then((data) => data as T);
+    });
+  }
 }
+
```
Comment (kleberbaum) [inline src/endpoints/scraper.ts]: Please remove newline.
```suggestion
```

File: src/session/sessions.ts [hunk 8]
```diff
@@ -218,11 +269,12 @@ class SnekSession extends CookieSession {
 
       if (!user) {
         /* Authenticate anonymous user */
-        authData = (await this.tasks.auth.anon()).data?.auth;
+
```
Comment (kleberbaum) [inline src/session/sessions.ts]: Please remove newline.
```suggestion
```

File: src/templates/snek/types.ts [hunk 9]
```diff
@@ -0,0 +1,47 @@
+//#region > General Types
+export { ApolloResult } from "../../endpoints/index";
+export { User } from "../../session";
+
+export type TaskTypes = "query" | "mutation";
+//#endregion
+
+//#region > Graphql Response Types for Authentication Related Requests
```
Comment (kleberbaum) [inline src/templates/snek/types.ts]: Extremely long region name.

**Review by kleberbaum — CHANGES_REQUESTED** [review]
Should be ok to merge after a few changes.

**Review by kleberbaum — APPROVED** [review]
Should be ok to merge.

**Pull request merged**